scripts/perfmon

```diff
@@ -938,20 +936,20 @@ def update_all_xmlconfigs(session):
     for recs in (all_host_recs, all_vm_recs, all_sr_recs):
         all_otherconfigs.update([
                 (recs[ref]['uuid'], recs[ref]['other_config'])
-                for ref in recs.keys()
+                for ref in list(recs.keys())
```

Comment:
It appears to be sufficient to write it this way:
```suggestion
                for ref in recs
```

Example code:
```py
recs = {"key1": {"attr": "item1"}, "key2": {"attr":"item2"}}
print("Output:", [(ref, recs[ref]["attr"]) for ref in recs])
Output: [('key1', 'item1'), ('key2', 'item2')]
```

You find a good explanation and many conformations here:
https://stackoverflow.com/questions/27476079/why-does-2to3-change-mydict-keys-to-listmydict-keys
Quote:
> In Python 3, the .keys() method returns a view object rather than a list, for efficiency's sake.
>
> In the iteration case, this doesn't actually matter, but where it would matter is if you were doing something like foo.keys()[0] - you can't index a view. Thus, 2to3 always adds an explicit list conversion to make sure that any potential indexing doesn't break.
>
> You can manually remove the list() call anywhere that a view would work fine; 2to3 just isn't smart enough to tell which case is which.

The same change has been requested by `rosslagerwall` in other reviews and other I sat another pull request which directly changed the code this way.

This applies to all uses of `for key in dict.keys()` in this PR.

In my testing, I also saw that the same applies to all uses of `for key in dict.items()` and `for key in dict.values()` where these forms appear to be sufficient.

It's just that 2to3 does not check if `dict.keys()` / `.items()` / `.values()` is used as `....()[0]`, and only in the latter case, the view needs to be converted into a list before the `[0]` or other direct indexing, not for the `for` loop.

Reply:
Got it. I'll check each change by 2to3.
Thank you so much!